ops/terraform/services/server/modules/bfd_server_asg/main.tf

```diff
@@ -180,89 +180,151 @@ resource "aws_autoscaling_group" "main" {
 
 ## Autoscaling Policies and Cloudwatch Alarms
 #
-resource "aws_autoscaling_policy" "high-cpu" {
-  name                      = "bfd-${local.env}-${var.role}-high-cpu-policy"
-  autoscaling_group_name    = aws_autoscaling_group.main.name
-  adjustment_type           = "ChangeInCapacity"
-  policy_type               = "StepScaling"
-  estimated_instance_warmup = var.asg_config.instance_warmup
-  metric_aggregation_type   = "Average"
+
+resource "aws_cloudwatch_metric_alarm" "filtered_networkin_low" {
+  alarm_name          = "bfd-${var.role}-${local.env}-networkin-low"
+  comparison_operator = "LessThanThreshold"
+  datapoints_to_alarm = 5
+  evaluation_periods  = 5
+  threshold           = 400000000
+  treat_missing_data  = "ignore"
+  alarm_actions       = [aws_autoscaling_policy.filtered_networkin_low_scaling.arn]
+
+  metric_query {
+    id          = "m1"
+    return_data = false
+
+    metric {
+      dimensions = {
+        AutoScalingGroupName = aws_autoscaling_group.main.name
+      }
+      metric_name = "NetworkIn"
+      namespace   = "AWS/EC2"
+      period      = 60
+      stat        = "Average"
+    }
+  }
+
+  metric_query {
+    id          = "m2"
+    return_data = false
+
+    metric {
+      dimensions = {
+        AutoScalingGroupName = aws_autoscaling_group.main.name
+      }
+      metric_name = "NetworkOut"
+      namespace   = "AWS/EC2"
+      period      = 60
+      stat        = "Average"
+    }
+  }
+
+  metric_query {
+    expression  = "IF(m2/m1 > 0.01, m1, 0)"
+    id          = "e1"
+    label       = "FilteredNetworkIn"
+    return_data = true
+  }
+}
+
+resource "aws_autoscaling_policy" "filtered_networkin_low_scaling" {
+  name                    = "bfd-${var.role}-${local.env}-networkin-low-scalein"
+  autoscaling_group_name  = aws_autoscaling_group.main.name
+  adjustment_type         = "ChangeInCapacity"
+  metric_aggregation_type = "Average"
+  policy_type             = "StepScaling"
 
   step_adjustment {
-    scaling_adjustment          = 1
-    metric_interval_lower_bound = 0.0
-    metric_interval_upper_bound = 15.0
+    metric_interval_upper_bound = "-3e+08"
+    scaling_adjustment          = -(length(var.env_config.azs) * 3)
   }
 
   step_adjustment {
-    scaling_adjustment          = 2
-    metric_interval_lower_bound = 15.0
-    metric_interval_upper_bound = 35.0
+    metric_interval_lower_bound = "-3e+08"
```

Comment:
extremely minor nit...but in some places we use numerics and other, scientific notation. Maybe have inline comment: i.e., e+08 #  100,000,000.

Reply:
> extremely minor nit...but in some places we use numerics and other, scientific notation. Maybe have inline comment: i.e., e+08 # 100,000,000.

I might hedge along with @mackec1–I'm _mostly_ ambivalent toward the scientific notation, but it might be helpful to contributors less familiar with the AWS landscape and the specifics of `terraform-provider-aws`. However, it might be more pressing to simply clarify/obviate the unit of the metrics evaluated here. Of course, [AWS/EC2/NetworkIn is in Bytes](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/viewing_metrics_with_cloudwatch.html#ec2-cloudwatch-metrics).

Reply:
Agreed. These values were included verbatim from `terraform import`ing then copying the `terraform state show` of the `import`ed autoscaling policy resources. I opted to keep it, as replacing the scientific notation strings with the corresponding numbers (i.e. `3e+08` to `300000000`) caused perpetual diffs in the `terraform apply` as the AWS API appears to represent large values in scientific notation and Terraform treats that as distinct from the numeric value.

I updated these definitions to use a combination of Terraform's [`format()`](https://developer.hashicorp.com/terraform/language/functions/format) and integer math to make these values more obvious. I also added comments indicating the unit of each metric threshold so that it's obvious these values represent _bytes_.

Reply:
> [...] caused perpetual diffs [...]

I'm sorry I've been a little disorganized lately... 

I assumed the perpetual diffs was the reason so I did a little research earlier this morning and found a corresponding GitHub issue in either `terraform` or the `terraform-aws-provider` that has been open since terraform 0.11 (❗).  If I recall correctly, a user weighed-in _somewhat_ recently to confirm it was still an issue for them as of 1.0.9. I must have closed the tab and couldn't be bothered to search again. No matter, I 'd just like to say that I believe you and that you're not alone 😉.